### ISSUE TITLE
yaml parse error not reporting file path

### ISSUE THREAD

**Issue by tcurdt** [11ty/eleventy#1152]
**Describe the bug**

Invalid yaml parsing does not report which file is the causing the problem.

**To Reproduce**

Just create some invalid yaml in the frontmatter.

**Expected behavior**

The exception should tell the exact path of the file that caused the problem.

**Environment:**

- macOS 10.15.3
- Eleventy Version 0.10.0

**Additional context**

```
Eleventy:EleventyErrorHandler Problem writing Eleventy templates: +0ms
  Eleventy:EleventyErrorHandler can not read a block mapping entry; a multiline key may not be an implicit key at line 7, column 1:
  Eleventy:EleventyErrorHandler     
  Eleventy:EleventyErrorHandler     ^
  Eleventy:EleventyErrorHandler 
  Eleventy:EleventyErrorHandler `YAMLException` was thrown: +0ms
  Eleventy:EleventyErrorHandler (error stack): YAMLException: can not read a block mapping entry; a multiline key may not be an implicit key at line 7, column 1:
  Eleventy:EleventyErrorHandler     
  Eleventy:EleventyErrorHandler     ^
  Eleventy:EleventyErrorHandler     at generateError (path/node_modules/js-yaml/lib/js-yaml/loader.js:167:10)
  Eleventy:EleventyErrorHandler     at throwError (path/node_modules/js-yaml/lib/js-yaml/loader.js:173:9)
  Eleventy:EleventyErrorHandler     at readBlockMapping (path/node_modules/js-yaml/lib/js-yaml/loader.js:1073:9)
  Eleventy:EleventyErrorHandler     at composeNode (path/node_modules/js-yaml/lib/js-yaml/loader.js:1359:12)
  Eleventy:EleventyErrorHandler     at readDocument (path/node_modules/js-yaml/lib/js-yaml/loader.js:1519:3)
  Eleventy:EleventyErrorHandler     at loadDocuments (path/node_modules/js-yaml/lib/js-yaml/loader.js:1575:5)
  Eleventy:EleventyErrorHandler     at load (path/node_modules/js-yaml/lib/js-yaml/loader.js:1596:19)
  Eleventy:EleventyErrorHandler     at Object.safeLoad (path/node_modules/js-yaml/lib/js-yaml/loader.js:1618:10)
  Eleventy:EleventyErrorHandler     at module.exports (path/node_modules/gray-matter/lib/parse.js:12:17)
  Eleventy:EleventyErrorHandler     at parseMatter (path/node_modules/gray-matter/index.js:109:17) +0ms

```

**Comment by zachleat** [comment]
Confirmed.

**Comment by zachleat** [comment]
Fix added, this will ship with the next release (probably 1.0)

**Comment by lefense** [comment]
Hi, I'm new to Eleventy but enjoying it a lot. However, I just came across this exact same issue and I'm on a newer version of Eleventy. (I'm also new to Github—please forgive me if I'm not supposed to post under a "closed" issue). Is the fix supposed to be in v0.11.1? I've copied the error message I received below. Thanks!


**Expected behavior**
Yaml exception does not indicate the exact path of the file causing the problem.

**Environment:**
macOS 10.15.4
Eleventy Version v0.11.1


```
Problem writing Eleventy templates: (more in DEBUG output)
> unexpected end of the stream within a flow collection at line 8, column 1:
    
    ^

`YAMLException` was thrown:
    YAMLException: unexpected end of the stream within a flow collection at line 8, column 1:
        ^
        at generateError ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:167:10)
        at throwError ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:173:9)
        at readFlowCollection ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:770:3)
        at composeNode ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1360:11)
        at readBlockMapping ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1036:16)
        at composeNode ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1359:12)
        at readDocument ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1525:3)
        at loadDocuments ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1588:5)
        at load ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1614:19)
        at Object.safeLoad ((path to my folder)/node_modules/js-yaml/lib/js-yaml/loader.js:1637:10)
Wrote 0 files in 0.24 seconds (v0.11.1)
Watching…
```

**Comment by pdehaan** [comment]
> Fix added, this will ship with the next release (probably 1.0)

@lefense I don't think this fix is included in eleventy v0.11.x.

This is the fix in the **"master"** branch:
https://github.com/11ty/eleventy/blob/904234193766aa9d9a7fb1e5037dea8838995a0f/src/TemplateContent.js#L85-L92

But this is the current code in the **"v0.11.x"** branch:
https://github.com/11ty/eleventy/blob/094c9851d98d77fdd6723673c71cff32de9a7f0a/src/TemplateContent.js#L82

So I think you'll either have to wait for eleventy v1.0, or see if you can install eleventy from GitHub instead of npm. <kbd>npm i github:11ty/eleventy</kbd> should do it, but it may or may not be what you want.

**Comment by lefense** [comment]
Ahh, ok, thanks @pdehaan! I understand now. I'll wait for the v1.0 release. And I was finally able to track down the file with my error in the meantime.


> > Fix added, this will ship with the next release (probably 1.0)
> 
>> @lefense I don't think this fix is included in eleventy v0.11.x.
>
> > So I think you'll either have to wait for eleventy v1.0, or see if you can install eleventy from GitHub instead of npm. npm i github:11ty/eleventy should do it, but it may or may not be what you want.
